src/HsmMoments.cc

```diff
@@ -86,6 +88,12 @@ void HsmMomentsAlgorithm::calculate(
     base::ShapeResult shapeResult;
     shapeResult.setShape(Ellipse(ellip, radius));
     source.set(_momentsKey, shapeResult);
+    if (addFlux) {
+        if (!_fluxKey.isValid()) {
+            throw LSST_EXCEPT(pex::exceptions::LogicError, "Flux Key not defined");
```

Comment:
Lower-case "key".

Should this be an `assert` instead of an exception? You explicitly created the key earlier, so if it doesn't exist then it's a programming error.